Packs/Inventa/Integrations/Inventa/Inventa.py

```diff
@@ -0,0 +1,465 @@
+                method="POST",
+                json_data=payload,
+                url_suffix="/dsr/api/dsar-management/tickets",
+                return_empty_response=True,
+                retries=5
+            )
+        return ticket_id
+
+    def get_ticket(self, ticket_id: Any) -> Dict:
+        result = self._http_request(
+            method="GET",
+            url_suffix=f"/dsr/api/dsar-management/tickets/{ticket_id}",
+            return_empty_response=True,
+            retries=5
+        )
+        return result
+
+    def get_dsar(self, ticket_id: Any) -> Dict:
+        return self._http_request(
+            method="GET",
+            url_suffix=f"/dsr/api/dsar-management/personal-data-usage/ticket/{ticket_id}",
+            return_empty_response=True,
+            retries=5
+        )
+
+
+''' HELPER FUNCTIONS '''
+
+
+def format_pii_entities(pii_entities):
+    entities_listed = [[subitem for subitem in pii_entities[item]] for item in pii_entities]
+    # return entities_listed
+    result = list()
+    for item in entities_listed:
+        result.extend(item)
+    result = {"entities": [item["entityName"] for item in result]}
+    return result
+
+
+def generate_datasubject_payload(pii_entities, **kwargs):
+    payload = []
+    # fill initial payload
+    for item in kwargs:
+        if item.upper() in pii_entities:
+            subpayload = {}
+            subpayload["piiEntityType"] = item.upper()
+            subpayload["piiEntityValue"] = kwargs[item]
+            payload.append(subpayload)
+    return payload
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def get_entities_command(client: Client) -> CommandResults:
+
+    pii_entities = client.get_entities()
+
+    result = format_pii_entities(pii_entities)
+    return CommandResults(
+        outputs_prefix="Inventa.Entities",
+        outputs_key_field="entity",
+        outputs=result
+    )
+
+
+def get_datasubjects_command(client: Client, **kwargs) -> CommandResults:
+    return CommandResults(outputs=client.get_datasubject(**kwargs),
+                          outputs_prefix="Inventa.DataSubjects",
+                          outputs_key_field="id")
+
+
+def get_datasubject_id_command(client: Client, **kwargs) -> CommandResults:
+    found_piis = client.get_datasubject(**kwargs)
+    datasubs = found_piis.get("dataSubjects", [])
+    if datasubs:
+        datasubject_id = datasubs[0].get("id", "")
+        return CommandResults(outputs={"datasubject_id": datasubject_id},
+                              outputs_prefix="Inventa.DataSubjects",
+                              outputs_key_field="datasubject_id")
+    else:
+        return CommandResults(outputs={"datasubject_id": 0},
+                              outputs_prefix="Inventa.DataSubjects",
+                              outputs_key_field="datasubject_id")
+
+
+def create_ticket_command(client: Client, reason: str, datasubject_id: int) -> CommandResults:
+    ticket_id = client.create_ticket(datasubject_id, reason)
```

Comment:
What happens if ticket_id is None?

Reply:
Basically, client.create_ticket() will raise an Exception in cases when the reason is missing or if the datasubject_id is missing or corrupted. Thus, the method will either return valid ticket_id, or it will raise an Exception and stop the playbook.